### DISCUSSION TITLE
[REJECTED] [test] attribute replacing fn test_f()

### DISCUSSION THREAD

**Discussion by igotfr** [vlang/v#11924] [Ideas]
```v
[test]
fn f() {
  assert 2 == 2
}

// instead
fn test_f() {
  assert 2 == 2
}
```

**Comment by JalonSolov** [comment]
There's no need for an extra attribute.  Naming the fn `test_<whatever>` works, and works well.  Go does the same thing with their test files, and it has worked for them for over a decade.

**Comment by igotfr** [comment]
> There's no need for an extra attribute.  Naming the fn `test_<whatever>` works, and works well.  Go does the same thing with their test files, and it has worked for them for over a decade.

maybe as an optional alternative

V has the "one way" philosophy, but add a bit of readabilty is always good. I don't know how it reflects on compiler or final performance, case be bad is better choose one between the alternatives, a open votation maybe

**Comment by islonely** [comment]
> There's no need for an extra attribute.  Naming the fn `test_<whatever>` works, and works well.  Go does the same thing with their test files, and it has worked for them for over a decade.

I don't think we should do something just because Go does. But I am in favor of `fn test_something`. I think attributes should be reserved for compiler related things. i.e. `[inline]` lets the compiler know to inline a function. `[test]` would just be about readability and doesn't tell the compiler anything.

**Comment by igotfr** [comment]
related:
https://github.com/vlang/rfcs/issues/13

**Comment by spytheman** [comment]
The `test_` prefix for test functions is preferable - it is easier and simpler to find such functions for both humans and for the test framework too.
